samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/OktaVerifyController.cs

```diff
@@ -0,0 +1,239 @@
+            if (!pollResponse.ContinuePolling)
+            {
+                switch (pollResponse?.AuthenticationStatus)
+                {
+                    case AuthenticationStatus.Success:
+                        ClaimsIdentity identity = await AuthenticationHelper.GetIdentityFromTokenResponseAsync(_idxClient.Configuration, pollResponse.TokenInfo);
+                        _authenticationManager.SignIn(new AuthenticationProperties(), identity);
+                        pollViewModel.Next = "/Home/Index";
+                        break;
+                    case AuthenticationStatus.PasswordExpired:
+                        pollViewModel.Next = "/Manage/ChangePassword";
+                        break;
+                    case AuthenticationStatus.AwaitingChallengeAuthenticatorSelection:
+                        Session["authenticators"] = ViewModelHelper.ConvertToAuthenticatorViewModelList(pollResponse.Authenticators);
+                        Session["isChallengeFlow"] = true;
+                        pollViewModel.Next = "/Manage/SelectAuthenticator";
+                        break;
+                    case AuthenticationStatus.AwaitingAuthenticatorEnrollment:
+                        Session["isChallengeFlow"] = false;
+                        Session["authenticators"] = ViewModelHelper.ConvertToAuthenticatorViewModelList(pollResponse.Authenticators);
+                        pollViewModel.Next = "/Manage/SelectAuthenticator";
+                        break;
+                    default:
+                        pollViewModel.Next = "/Account/Login";
+                        break;
+                }
+            }
+            
+            return Json(pollViewModel, JsonRequestBehavior.AllowGet);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> SelectAuthenticatorMethodAsync(OktaVerifySelectAuthenticatorMethodModel model)
+        {
+            var selectAuthenticatorOptions = new SelectOktaVerifyAuthenticatorOptions
+            {
+                AuthenticatorMethodType = model.MethodType,
+                AuthenticatorId = model.AuthenticatorId,
+            };
+            
+            try
+            {
+                var authnResponse = await _idxClient.SelectChallengeAuthenticatorAsync(selectAuthenticatorOptions,
+                    (IIdxContext)Session["IdxContext"]);
+
+                if (authnResponse.AuthenticationStatus == AuthenticationStatus.AwaitingAuthenticatorVerification)
+                {
+                    switch (model.MethodType)
+                    {
+                        case "totp":
+                            return View("EnterCode");
+                        case "push":
+                            return View("PushSent",
+                                new OktaVerifySelectAuthenticatorMethodModel());
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                ModelState.AddModelError("MethodType", e);
```

Comment:
We should return to the view with the model so the errors are shown in the UI. Something like this:
```
return View("SelectAuthenticatorMethod", model);
```